src/common/device_helpers.cuh

```diff
@@ -1009,6 +1044,29 @@ class AllReducer {
       dh::safe_cuda(cudaSetDevice(device_ordinals[i]));
       dh::safe_cuda(cudaStreamSynchronize(streams[i]));
     }
+  #endif
+  };
+
+  /**
+   * \fn  ncclUniqueId GetUniqueId()
+   *
+   * \brief Gets the Unique ID from NCCL to be used in setting up interprocess
+   * communication
+   *
+   * \return the Unique ID
+   */
+  ncclUniqueId GetUniqueId() {
+#ifdef XGBOOST_USE_NCCL
```

Comment:
I thought of doing this as well. Mostly, I was trying to be consistent with the other code. This, being an odd case because the return type is from NCCL.

Is there a particular reason XGB tends to declare empty functions?

Reply:
I don't think there is any particular reason.

Reply:
I shall wrap the function entirely then with a definition guard.

Reply:
+1.